examples/pipeline_tacotron2/inference.py

```diff
@@ -0,0 +1,294 @@
+    )
+    preprocessor.add_argument(
+        '--cmudict-root',
+        default="./",
+        type=str,
+        help='the root directory for storing CMU dictionary files'
+    )
+
+    audio = parser.add_argument_group('audio parameters')
+    audio.add_argument(
+        '--sample-rate',
+        default=22050,
+        type=int,
+        help='Sampling rate'
+    )
+    audio.add_argument(
+        '--n-fft',
+        default=1024,
+        type=int,
+        help='Filter length for STFT'
+    )
+    audio.add_argument(
+        '--n-mels',
+        default=80,
+        type=int,
+        help=''
+    )
+    audio.add_argument(
+        '--mel-fmin',
+        default=0.0,
+        type=float,
+        help='Minimum mel frequency'
+    )
+    audio.add_argument(
+        '--mel-fmax',
+        default=8000.0,
+        type=float,
+        help='Maximum mel frequency'
+    )
+
+    # parameters for WaveRNN
+    wavernn = parser.add_argument_group('WaveRNN parameters')
+    wavernn.add_argument(
+        '--wavernn-checkpoint-name',
+        default="wavernn_10k_epochs_8bits_ljspeech",
+        choices=list(_MODEL_CONFIG_AND_URLS.keys()),
+        help="Select the WaveRNN checkpoint."
+    )
+    wavernn.add_argument(
+        "--wavernn-loss",
+        default="crossentropy",
+        choices=["crossentropy"],
+        type=str,
+        help="The type of loss the WaveRNN pretrained model is trained on.",
+    )
+    wavernn.add_argument(
+        "--wavernn-no-batch-inference",
+        default=False,
+        action="store_true",
+        help="Don't use batch inference for WaveRNN inference."
+    )
+    wavernn.add_argument(
+        "--wavernn-no-mulaw",
+        default=False,
+        action="store_true",
+        help="Don't use mulaw decoder to decode the signal."
+    )
+    wavernn.add_argument(
+        "--wavernn-batch-timesteps",
+        default=11000,
+        type=int,
+        help="The time steps for each batch. Only used when batch inference is used",
+    )
+    wavernn.add_argument(
+        "--wavernn-batch-overlap",
+        default=550,
+        type=int,
+        help="The overlapping time steps between batches. Only used when batch inference is used",
+    )
+
+    return parser
+
+
+def unwrap_distributed(state_dict):
+    r"""torch.distributed.DistributedDataParallel wraps the model with an additional "module.".
+    This function unwraps this layer so that the weights can be loaded on models with a single GPU.
+
+    Args:
+        state_dict: Original state_dict.
+
+    Return:
+        unwrapped_state_dict: Unwrapped state_dict.
+    """
+
+    return {k.replace('module.', ''): v for k, v in state_dict.items()}
+
+
+def main(args):
+    torch.manual_seed(0)
+    random.seed(0)
+    np.random.seed(0)
+
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    n_symbols = len(get_symbol_list(args.text_preprocessor))
+    text_preprocessor = partial(
```

Comment:
Similar to https://github.com/pytorch/audio/pull/1668/files#r686174534, turning this into a class will make the code more maintainable.

Reply:
I'll submit a separate PR for this. I've created an issue #1710 to keep track of this.